samples/GalleryApp/MainPage.xaml.cs

```diff
@@ -67,5 +83,25 @@ public void NavigateToSkillPage(Skill skill)
         {
             this.Frame.Navigate(skill.PageType);
         }
+
+        public static async Task<List<SkillCategory>> GetCategoriesAsync()
+        {
+            await _semaphore.WaitAsync();
```

Comment:
>await _semaphore.WaitAsync(); [](start = 12, length = 29)

@Louis @Mariana I referenced the Microsoft Community Toolkit's implementation on this function, and am not sure if using thread is necessary here because there's only one task (loading the file)?